pyquil/api/_wavefunction_simulator.py

```diff
@@ -224,77 +233,74 @@ def augment_program_with_memory_values(
 
         if len(memory_map.keys()) == 0:
             return quil_program
-        elif isinstance(list(memory_map.keys())[0], MemoryReference):
-            warn(
-                "Use of memory_map values of type Dict[MemoryReference, Any] have been "
-                "deprecated.  Please use Dict[str, List[Union[int, float]]], as with "
-                "QuantumComputer.run ."
-            )
-            for k, v in memory_map.items():
-                p += MOVE(k, v)
         elif isinstance(list(memory_map.keys())[0], str):
             for name, arr in memory_map.items():
                 for index, value in enumerate(arr):
-                    p += MOVE(MemoryReference(cast(str, name), offset=index), value)
+                    p += MOVE(MemoryReference(name, offset=index), value)
         else:
             raise TypeError("Bad memory_map type; expected Dict[str, List[Union[int, float]]].")
 
         p += quil_program
 
         return percolate_declares(p)
 
+    def _run_and_measure_request(
+        self,
+        *,
+        quil_program: Program,
+        qubits: Sequence[int],
+        trials: int,
+    ) -> RunAndMeasureProgramRequest:
+        if not quil_program:
+            raise ValueError(
+                "You have attempted to run an empty program."
+                " Please provide gates or measure instructions to your program."
+            )
+
+        if not isinstance(quil_program, Program):
+            raise TypeError("quil_program must be a Quil program object")
```

Comment:
include the actual type?

Reply:
meaning the type of `quil_program`

and also replace "Quil program object" with "Program object"

Reply:
Gotcha, this was c/p'd from QVMConnection, but will update. Good catch